main.go

```diff
@@ -57,15 +56,37 @@ type ProcDiskIOStat struct {
 }
 
 // MozProcessStat combines existing structs into one.
+// CPU is a map so that we can store per-process data.
 type MozProcessStat struct {
-	Timestamp    int64              `json:"timestamp"`
-	Memory       ProcMemoryInfoStat `json:"memory"`  // all uint64
-	CPU          ProcCPUStat        `json:"cpu"`     // all float64
-	DiskIO       ProcDiskIOStat     `json:"disk"`    // all uint64
-	NetworkIO    ProcNetworkIOStat  `json:"network"` // all uint64
-	UsedPercent  float64            `json:"system_memory_used_percent"`
-	ProcessCount int                `json:"process_count"`
-	ThreadCount  int32              `json:"thread_count"`
+	Memory    ProcMemoryInfoStat `json:"memory"`  // all uint64
+	CPU       ProcCPUStat        `json:"cpu"`     // all float64
+	DiskIO    ProcDiskIOStat     `json:"disk"`    // all uint64
+	NetworkIO ProcNetworkIOStat  `json:"network"` // all uint64
+}
+
+// MozCollectedStat combines existing structs into one.
+// CPU is a map so that we can store per-process data.
+type MozCollectedStat struct {
+	Timestamp       int64                     `json:"timestamp"`
+	Processes       map[int32]*MozProcessStat `json:"process_stats"`
+	AvailableMemory uint64                    `json:"available_memory"` // bytes
+	UsedPercent     float64                   `json:"system_memory_used_percent"`
```

Comment:
I should give this a better name since we're also collecting cpu percentages now

Reply:
Makes sense. I think we can name them `CPUUsedPercent` and `MemoryUsedPercent` or so. After all, these will be among the things we'll look the most at in our stats.